### PR TITLE
Gnomon decompose API

### PR DESCRIPTION
**Pull request by graft**
A start on the decompose API - this does the basic thing, but still lacks the reporting on identifier creation, in the Gnomon identifier tables and in the library tables. Also the front end is not wired up yet to use this.

### CODE REVIEW

File: magma/lib/magma/gnomon.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+require_relative './gnomon/rule_errors'
+require_relative './gnomon/rule_definition'
```
Comment (coleshaw) [inline magma/lib/magma/gnomon.rb:2]: Nice, thanks for separating these out more cleanly!

**Review by coleshaw — APPROVED** [review]
Nice! I started a "counter" API that builds off of this, but haven't been able to make much progress...

**Comment by graft** [comment]
Ok, bunch of fixes, I think it should be ready to go, barring a few tests.

This is maybe a bit over the top:
create

**Comment by dtm2451** [comment]
> This is maybe a bit over the top

bahahaha, but I love it!

**Comment by coleshaw** [comment]
That is psychadelic!

**Comment by coleshaw** [comment]
Can you make it generate fireworks and sparks / smoke on click?

**Pull request merged**